crates/sui-indexer-alt/src/handlers/obj_info.rs

```diff
@@ -105,10 +76,88 @@ impl Processor for ObjInfo {
 #[async_trait::async_trait]
 impl Handler for ObjInfo {
     async fn commit(values: &[Self::Value], conn: &mut db::Connection<'_>) -> Result<usize> {
+        let stored = values
+            .iter()
+            .map(|v| v.try_into())
+            .collect::<Result<Vec<StoredObjInfo>>>()?;
         Ok(diesel::insert_into(obj_info::table)
-            .values(values)
+            .values(stored)
             .on_conflict_do_nothing()
             .execute(conn)
             .await?)
     }
 }
+
+pub(crate) enum ProcessedObjInfoUpdate {
+    Insert(Object),
+    Delete(ObjectID),
+}
+
+pub(crate) struct ProcessedObjInfo {
+    pub cp_sequence_number: u64,
+    pub update: ProcessedObjInfoUpdate,
+}
```

Comment:
nit: Could you move the top-level elements around for consistent file order? types at the top, then impl blocks, then trait impls, then free functions.

nit (optional): I think I started with something quite similar to this pattern for `StoredObjectUpdate`, but ended up going for something more like this (not using a special enum type and pulling the `object_id` into the outer struct) as it made some things neater:

```rust
pub(crate) struct ProcessedObjInfo {
    pub object_id: ObjectID,
    pub cp_sequence_number: u64,
    pub update: Option<Object>
}
```

Reply:
> nit: Could you move the top-level elements around for consistent file order? types at the top, then impl blocks, then trait impls, then free functions.

hmm isn't it always better for impls to be right next to its struct definition? Otherwise it makes it diffult to locate them.

Reply:
This ordering of elements in the file helps in two ways: 

- It's consistent across the codebase (in this case, in `sui-indexer-alt`/`sui-graphql-rpc`/`sui-package-resolver`) which helps to orient quickly (the reason I noticed this was that I looked for `ProcessedObjInfo` at the top of the file when I saw it mentioned and I couldn't find it there).
- It's a forcing function for splitting up files, when they get large enough that a type is far away from its impl block -- otherwise it becomes very easy to concatenate together multiple clusters of types with their impl blocks in a file that grows arbitrarily long without noticing, because you view each cluster as its own unit. It only becomes a problem when someone tries to take a holistic view of the file, and that person struggles.